doc/Building.md

```diff
@@ -0,0 +1,199 @@
+# Setting up a build environment for KOReader
+
+These instructions are intended to build the emulator in Linux and MacOS. Windows users are suggested to develop in a [Linux VM](https://www.howtogeek.com/howto/11287/how-to-run-ubuntu-in-windows-7-with-vmware-player/) or using the [Windows Subsystem for Linux](https://en.wikipedia.org/wiki/Windows_Subsystem_for_Linux).
+
+If you only want to work with Lua frontend stuff, you can grab the AppImage and
+run it with `--appimage-extract`.
+
+You can skip most of the following instructions if desired, and use our premade Docker image instead. In that case the only requirements are Git and Docker. See [the virtual development environment README](https://github.com/koreader/virdevenv) for more information.
+
+## Prerequisites
+
+To get and compile the source you must have `patch`, `wget`, `unzip`, `git`,
+`cmake` and `luarocks` installed, as well as a version of `autoconf`
```

Comment:
luarocks seems to have become a minor issue in the last year (regardless whether we're talking Mac OS or bleeding user distros like Arch & Gentoo). For Docker we roll out our own, see https://github.com/koreader/virdevenv/blob/9b41d9f36d9b20fdf9b58d04104cc9133fc15ce5/docker/ubuntu/baseimage/install_luarocks.sh#L7-L13

We should probably at least add a little note to that script for the moment, although perhaps ideally e'd move away from system luarocks since its config files/options and such are annoyingly variable across versions.

Reply:
We can address this issue here too: https://github.com/koreader/koreader/issues/2798#issuecomment-299973036